tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -233,6 +290,25 @@ def test_metadata_upgrade(self, from_version, to_version):
 
         self.driver.stop()
 
+    def update_leader(self):
+        self.leader = None
+        retries = 10
+        while retries > 0:
+            for p in self.processors:
+                found = list(p.node.account.ssh_capture("grep \"Finished assignment for group\" %s" % p.LOG_FILE, allow_fail=True))
+                if len(found) == self.leader_counter[p] + 1:
+                    self.leader = p
+                    self.leader_counter[p] = self.leader_counter[p] + 1
```

Comment:
Why the `+ 1`? Is that for the leader to kick off version probing with a future version?

Reply:
If a `processor` is the leader, it will print `Finished assignment for group` in the log. A `processor` can be the leader multiple times, and thus we count how many of those lines we have seen in `leader_counter[p]`.  To identify the new leader, it's count must increase by exactly one. We increase by one as we have found one occurrence as expected.